File: shared_resources/src/main/java/com/thewizrd/shared_resources/helpers/WearConnectionStatus.java
package com.thewizrd.shared_resources.helpers;

import android.util.SparseArray;

public enum WearConnectionStatus {
    DISCONNECTED(0),
    CONNECTING(1),
    APPNOTINSTALLED(2),
    CONNECTED(3);

    private final int value;

    public int getValue() {
        return value;
    }

    private WearConnectionStatus(int value) {
        this.value = value;
    }

    private static SparseArray<WearConnectionStatus> map = new SparseArray<>();

    static {
        for (WearConnectionStatus connectionStatus : values()) {
            map.put(connectionStatus.value, connectionStatus);
        }
    }

    public static WearConnectionStatus valueOf(int value) {
        return map.get(value);
    }
}

File: shared_resources/src/main/java/com/thewizrd/shared_resources/helpers/WearConnectionStatus.kt
package com.thewizrd.shared_resources.helpers

import android.util.SparseArray

enum class WearConnectionStatus(val value: Int) {
    DISCONNECTED(0),
    CONNECTING(1),
    APPNOTINSTALLED(2),
    CONNECTED(3);

    companion object {
        private val map = SparseArray<WearConnectionStatus>()

        fun valueOf(value: Int): WearConnectionStatus {
            return map[value]
        }

        init {
            for (connectionStatus in values()) {
                map.put(connectionStatus.value, connectionStatus)
            }
        }
    }
}